src/Integration/RuleSets/SonarQubeRuleSetProvider.cs

```diff
@@ -0,0 +1,169 @@
+            {
+                return null;
+            }
+
+            var ruleset = ConvertToCodeAnalysisRuleSet(roslynProfileExporter);
+
+            return ConvertToSonarRuleSet(ruleset, language);
+        }
+
+        private async Task<RoslynExportProfileResponse> GetSonarQubeRuleSet(BoundSonarQubeProject project, Language language)
+        {
+            var serverLanguage = language.ToServerLanguage();
+
+            var qualityProfileInfo = await SafeServiceCall(() => this.sonarQubeService.GetQualityProfileAsync(
+                project.ProjectKey, project.Organization?.Key, serverLanguage, CancellationToken.None));
+            if (qualityProfileInfo == null)
+            {
+                this.logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
+                   string.Format(Strings.CannotDownloadQualityProfileForLanguage, language.Name)));
+                return null;
+            }
+
+            var roslynProfileExporter = await SafeServiceCall(() => this.sonarQubeService.GetRoslynExportProfileAsync(
+                qualityProfileInfo.Name, project.Organization?.Key, serverLanguage, CancellationToken.None));
+            if (roslynProfileExporter == null)
+            {
+                this.logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
+                    string.Format(Strings.QualityProfileDownloadFailedMessageFormat, qualityProfileInfo.Name,
+                        qualityProfileInfo.Key, language.Name)));
+                return null;
+            }
+
+            return roslynProfileExporter;
+        }
+
+        private async Task<T> SafeServiceCall<T>(Func<Task<T>> call)
+        {
+            try
+            {
+                return await call();
+            }
+            catch (HttpRequestException e)
+            {
+                // For some errors we will get an inner exception which will have a more specific information
+                // that we would like to show i.e.when the host could not be resolved
+                var innerException = e.InnerException as System.Net.WebException;
+                this.logger.WriteLine(Strings.SonarQubeRequestFailed, e.Message, innerException?.Message);
+            }
+            catch (TaskCanceledException)
+            {
+                // Canceled or timeout
+                this.logger.WriteLine(Strings.SonarQubeRequestTimeoutOrCancelled);
+            }
+            catch (Exception ex)
+            {
+                this.logger.WriteLine(Strings.SonarQubeRequestFailed, ex.Message, null);
+            }
+
+            return default(T);
+        }
+
+        private RuleSet ConvertToCodeAnalysisRuleSet(RoslynExportProfileResponse roslynProfileExporter)
+        {
+            var tempRuleSetFilePath = Path.GetTempFileName();
+            File.WriteAllText(tempRuleSetFilePath, roslynProfileExporter.Configuration.RuleSet.OuterXml);
+
+            return RuleSet.LoadFromFile(tempRuleSetFilePath);
```

Comment:
It's unfortunate we can only load the RuleSet from a file.
We could try to clean up the temp file when we're done.